woocommerce/woocommerce
# Post test result summary in workflow run and in PR comment using github-actions[bot]

### Changes proposed in this Pull Request:

This PR posts a summary of test results in the following places:

1. In the workflow run, like so:
  <img width="928" alt="Screen Shot 2022-07-04 at 8 08 31 PM" src="https://user-images.githubusercontent.com/4509348/177153590-54d4c3cd-1ad0-4429-9696-e22730af525f.png">

2. In a PR comment, effectively replacing the comment that `botwoo` creates. This time, the new PR comment is by `github-actions[bot]`.
  <img width="929" alt="Screen Shot 2022-07-04 at 8 08 46 PM" src="https://user-images.githubusercontent.com/4509348/177153723-72932d0e-85bf-44d3-af15-17879f1a3f03.png">

This will work on both Playwright and Puppeteer E2E tests.

Closes https://github.com/woocommerce/woocommerce-quality/issues/305.

### How to test the changes in this Pull Request:

1. Duplicate this PR.
2. In the duplicate PR, toggle the `E2E_PLAYWRIGHT` flag on by setting this line of code to `${{ true }}` in the `pr-build-and-e2e-tests.yml` workflow file:
  https://github.com/woocommerce/woocommerce/blob/3e6196a31229bbb02ddd94acfab85ad43cbf47e5/.github/workflows/pr-build-and-e2e-tests.yml#L11
3. Wait for the new "Post test results" job to finish.
    <img width="662" alt="Screen Shot 2022-07-04 at 8 11 28 PM" src="https://user-images.githubusercontent.com/4509348/177154716-23d53645-1493-4ea3-b88c-441cac9bb05f.png">
4. Verify that you should see the "Test Results Summary" at the end of the workflow run page, like so:
    ![Screenshot 2022-07-04 at 20-30-58 Post test result summary in workflow run and in PR comment using github-actions bot · woocommerce_woocommerce@e38962a](https://user-images.githubusercontent.com/4509348/177155780-96493588-1dd2-4330-9241-62cac1db73c8.png)
5. Go to the PR page and verify that you no longer see `botwoo`'s PR comment. Instead, you would see `github-actions[bot]`'s test results summary like so:
    <img width="929" alt="Screen Shot 2022-07-04 at 8 08 46 PM" src="https://user-images.githubusercontent.com/4509348/177156075-dd2b6c04-aa8f-4dd5-9d90-3abd75e7da7d.png">
6. Switch the `E2E_PLAYWRIGHT` toggle to `false` so that this time, the Puppeteer e2e tests would run.
7. Again, wait for the "Post test results" job to finish.
8. Again, verify that you see the "Test Results Summary" at the end of this Puppeteer workflow run.
9. When you go back to the PR page in GitHub, notice that the PR comment by `github-actions[bot]` earlier was updated -- it's now showing the new Commit SHA and the new values in the `E2E Tests` row.
10. Verify that the links in the test summary are working. The link to the full E2E test report points to the WooCommerce Test Reports Dashboard for the meantime.

### FOR PR REVIEWER ONLY:
